src/statemachine.rs

```diff
@@ -874,6 +876,8 @@ impl StateMachine {
                         }
                     }
 
+                    send_status(&status, crate::StatusUpdate::PresenceRequired);
```

Comment:
Same idea here, unless there's a reason this needs to happen before processing the allow list?